### PR TITLE
feat(nameservice): implemented nameservice-crypto, nameservice-transactions and nameservice-api

### PR DESCRIPTION
**Pull request by KovacZan**
## Summary

Implemented nameservice.
It adds a name to wallet that sends the transaction, which can be then used for api searching.

I have implemented the following packages:
- nameservice-crypto
- nameservice-transactions
- nameservice-api

## Checklist

- [ ] Documentation _(if necessary)_
- [x] Tests _(if necessary)_
- [x] Ready to be merged

### CODE REVIEW

**Comment by kristjank** [comment]
👏

File: packages/nameservice-crypto/src/enums.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { defaults } from "./defaults";
+
+export enum NameServiceTransactionTypes {
+    Nameservice = 0,
+}
+
+export const NameServiceTransactionGroup = defaults.nameserviceTypeGroup;
+
+// parse to string
+const namesericeStaticFee: any = defaults.nameserviceStaticFee.toString();
```
Comment (amacar) [inline packages/nameservice-crypto/src/enums.ts:10]: typo, change to `nameserviceStaticFee` also change type to string

File: packages/nameservice-crypto/package.json [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        "test": "jest",
+        "test:watch": "jest --watchAll",
+        "coverage:report": "codecov",
+        "test:unit": "jest __tests__/unit/** --forceExit",
+        "test:unit:coverage": "jest __tests__/unit --coverage",
+        "publish:beta": "yarn build && yarn npm publish --tag beta --access public --tolerate-republish"
+    },
+    "dependencies": {
+        "@arkecosystem/crypto": "^3.0.0-next",
+        "@protokol/utils": "^1.0.0-beta.37",
```
Comment (amacar) [inline packages/nameservice-crypto/package.json:45]: Use latest version (.38).

File: packages/nameservice-api/src/resources/nameservice.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+     * @param {*} resource
+     * @returns {object}
+     * @memberof Resource
+     */
+    public transform(resource): object {
+        AppUtils.assert.defined<string>(resource.senderPublicKey);
+
+        const sender: string = this.walletRepository.findByPublicKey(resource.senderPublicKey).address;
+
+        return {
```
Comment (amacar) [inline packages/nameservice-api/src/resources/nameservice.ts:32]: do we need all this data in the nameservice response?

**Review by amacar — CHANGES_REQUESTED** [review]

**Review by amacar — APPROVED** [review]

**Pull request merged**